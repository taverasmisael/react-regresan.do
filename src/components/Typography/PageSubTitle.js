import React from 'react'
import PropTypes from 'prop-types'
import Typography from 'material-ui/Typography'

const PageSubTitle = ({ children, color = 'inherit', truncate = false, align = 'left' }) => (
  <Typography gutterBottom type="subheading" color={color} align={align} noWrap={truncate}>
    {children}
  </Typography>
)

PageSubTitle.propTypes = {
  children: PropTypes.string,
  align: PropTypes.oneOf(['inherit', 'left', 'center', 'right', 'justify']),
  color: PropTypes.oneOf(['inherit', 'primary', 'secondary', 'accent', 'error', 'default']),
  truncate: PropTypes.bool
}

export default PageSubTitle
