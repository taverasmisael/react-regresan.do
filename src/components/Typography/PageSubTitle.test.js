import React from 'react'
import { shallow } from 'enzyme'

import PageSubTitle from './PageSubTitle'

describe('components/Typography/PageSubTitle', () => {
  describe('rendering', () => {
    it('renders without crashing', () => {
      shallow(<PageSubTitle />)
    })
    it('matches snapshot', () => {
      expect(<PageSubTitle />).toMatchSnapshot()
      expect(<PageSubTitle>Hello World</PageSubTitle>).toMatchSnapshot()
    })
  })

  describe('variants', () => {
    it('text display correctly', () => {
      const component = shallow(<PageSubTitle>Subtitle</PageSubTitle>)
      expect(component.props().type).toBe('subheading')
    })
  })
})
