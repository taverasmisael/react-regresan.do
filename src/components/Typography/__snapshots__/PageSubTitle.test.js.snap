// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Typography/PageSubTitle rendering matches snapshot 1`] = `<PageSubTitle />`;

exports[`components/Typography/PageSubTitle rendering matches snapshot 2`] = `
<PageSubTitle>
  Hello World
</PageSubTitle>
`;
